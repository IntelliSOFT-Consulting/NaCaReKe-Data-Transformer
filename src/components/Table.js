import React, { useState, useEffect } from 'react';
import { Table } from 'antd';

export default function DataTable({ data, cols }) {
  const [rowData, setRowData] = useState(null);
  const [colData, setColData] = useState(null);

  const columns = cols =>
    cols &&
    cols.map((item, i) => ({
      title: item.name,
      dataIndex: item.key,
      key: item.key,
    }));

  const rowDatas = data =>
    data.map((r, i) => {
      const obj = {};
      cols.forEach(c => {
        obj[c.key] =
          r[c.key] || typeof r[c.key] == 'boolean'
            ? r[c.key].toString()
            : r[c.key];
        obj.key = i;
      });
      return obj;
    });

  useEffect(() => {
    if (cols) {
      setColData(columns(cols));
    }
    if (data.length > 0) {
      setRowData(rowDatas(data));
    }
  }, [data, cols]);

  return (
    <div data-testid='table' className='full-width'>
      <Table dataSource={rowData} columns={colData} />
    </div>
  );
}
