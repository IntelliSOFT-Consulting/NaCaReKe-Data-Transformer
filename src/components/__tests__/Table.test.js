import Table from '../Table';

const rowData = [
  ['OrgUnit', 'Incid', 'IncidType', 'IncidDesc', 'IncidSeverity'],
  ['OrgUnit', 'Incid', 'IncidType', 'IncidDesc', 'IncidSeverity'],
];

const colData = [
  { key: 0, name: 'A' },
  { key: 1, name: 'B' },
  { key: 2, name: 'C' },
  { key: 3, name: 'D' },
  { key: 4, name: 'E' },
];

it('renders correctly', () => {
  const tree = shallow(<Table data={rowData} cols={colData} />);
  expect(tree).toMatchSnapshot();
});

it('renders the correct rows of data', () => {
  const wrapper = render(<Table data={rowData} cols={colData} />);
  expect(wrapper.find('tr').length).toEqual(2);
});
