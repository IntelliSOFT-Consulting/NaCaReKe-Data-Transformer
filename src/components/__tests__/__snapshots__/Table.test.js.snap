// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="full-width"
  data-testid="table"
>
  <Table
    columns={null}
    dataSource={null}
    rowKey="key"
  />
</div>
`;
